distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso

```diff
@@ -602,7 +598,7 @@ type Vector a
         arr = Array.new (self_len + that.length)
         Array.copy self.to_array 0 arr 0 self_len
         Array.copy that.to_array 0 arr self_len that.length
```

Comment:
These two lines are going to perform 2 copies if the `that` was not an array, right?
Are these copies even necessary? If I understand the new design correctly, the new `Vector` type behaves like a Polyglot Array from Polyglot's perspective, so it can be passed without conversion into Java where `List<>` or `T[]` are expected, right? 

If so, I'd assume _almost all_ `vec.to_array` instances could be replaced with just `vec`. Is my assumption correct?

This is pretty important for me to understand right - because from what I see, now all calls to our Java helpers that had a Vector, when they call `to_array` - they will perform a copy - just to be able to pass the Vector to Java. BUT in most cases the methods treat the input array as immutable, so they should be able to work with the input data without copying (and that's what was happening in the earlier times when `to_array` was not copying data). It seems quite undesirable to me to make so many polyglot calls copy the array unnecessarily.

Reply:
`Vector.to_array` has been modified in f0338c5 to just return the underlying _array storage_. No copy is performed anymore.